apps/src/sites/studio/pages/foorm/forms/editorHelpers.js

```diff
@@ -0,0 +1,52 @@
+import {getStore} from '@cdo/apps/redux';
+import initializeCodeMirror from '@cdo/apps/code-studio/initializeCodeMirror';
+import {
+  setQuestions,
+  setHasJSONError
+} from '@cdo/apps/code-studio/pd/foorm/editor/foormEditorRedux';
+import _ from 'lodash';
+
+let codeMirror;
+
+function populateCodeMirror() {
+  const codeMirrorArea = document.getElementsByTagName('textarea')[0];
+
+  const updateQuestions = questions => {
+    getStore().dispatch(setQuestions(questions));
+    getStore().dispatch(setHasJSONError(false));
+  };
+
+  function onCodeMirrorChange(editor) {
+    try {
+      const questions = JSON.parse(editor.getValue());
+      updateQuestions(questions);
+    } catch (e) {
+      // There is a JSON error.
+      getStore().dispatch(setQuestions({}));
+      getStore().dispatch(setHasJSONError(true));
+    }
+  }
+
+  codeMirror = initializeCodeMirror(codeMirrorArea, 'application/json', {
+    callback: _.debounce(onCodeMirrorChange, 250)
+  });
+}
+
+function resetCodeMirror(json) {
+  if (codeMirror) {
+    codeMirror.setValue(JSON.stringify(json, null, 2));
+    getStore().dispatch(setHasJSONError(false));
+  }
+}
+
+const confirmNoUnsavedChanges = evt => {
+  let storeState = getStore().getState().foorm;
+  if (
+    storeState.hasJSONError ||
+    !_.isEqual(storeState.lastSavedQuestions, storeState.questions)
+  ) {
+    return 'Are you sure you want to exit? You may have unsaved changes.';
+  }
+};
+
+export {populateCodeMirror, resetCodeMirror, confirmNoUnsavedChanges};
```

Comment:
in LP code, we usually export things as part of their definition, e.g. `export function resetCodeMirror(json)...`. unless doing it this way is a PLC convention, i think we try to be consistent about that

Reply:
No preference, switched to LP standard.